cpp/src/io/orc/stats_enc.cu

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "orc_common.h"
+#include "orc_gpu.h"
+#include <io/utilities/block_utils.cuh>
+
+namespace cudf {
+namespace io {
+namespace orc {
+namespace gpu {
+
+/**
+ * @brief Initializes statistics groups
+ *
+ * @param[out] groups Statistics groups
+ * @param[in] cols Column descriptors
+ * @param[in] num_columns Number of columns
+ * @param[in] num_rowgroups Number of rowgroups
+ * @param[in] row_index_stride Rowgroup size in rows
+ *
+ **/
+constexpr unsigned int log2_init_thread_per_group = 5;
```

Comment:
Why bother with the log2? Just `init_thread_per_group = 32` and `init_groups_per_block = 4` would be fine. Then you can use multiply and divide instead of shifts. Fewer constexpr variables, shorter names, clearer code.

Reply:
It makes it clear that it must be a power of 2

Reply:
It's a matter of maintainability and reducing technical debt.

Reply:
[Affirming the consequent](https://en.wikipedia.org/wiki/Affirming_the_consequent)